gcp-observability/src/main/java/io/grpc/gcp/observability/GcpObservability.java

```diff
@@ -139,14 +145,23 @@ private void setProducer(
         clientInterceptors, serverInterceptors, tracerFactories);
   }
 
-  private void registerStackDriverExporter(String projectId) throws IOException {
+  private void registerStackDriverExporter(String projectId, Map<String, String> customTags)
+      throws IOException {
     if (config.isEnableCloudMonitoring()) {
       RpcViews.registerAllGrpcViews();
       StackdriverStatsConfiguration.Builder statsConfigurationBuilder =
           StackdriverStatsConfiguration.builder();
       if (projectId != null) {
         statsConfigurationBuilder.setProjectId(projectId);
       }
+      if (customTags != null) {
+        Map<LabelKey, LabelValue> constantLabels = new HashMap<>();
+        for (Map.Entry<String, String> mapEntry : customTags.entrySet()) {
+          constantLabels.put(LabelKey.create(mapEntry.getKey(), mapEntry.getKey()),
+              LabelValue.create(mapEntry.getValue()));
+        }
```

Comment:
It needs to be this: do you agree?

```
       Map<LabelKey, LabelValue> constantLabels = customTags.entrySet().stream()
                .collect(Collectors.toMap(k -> LabelKey.create(k.getKey(), k.getKey()),
                   v -> LabelValue.create(v.getValue())));
```

because k and v are Map.Entry (and not Map.Key and Map.Value)

Reply:
Ah yes - I did say "something like this" ;-)

Reply:
fixed, PTAL